### PR TITLE
Add Http proxy test scenario related to auto-attaching subscription to content host.

### PR DESCRIPTION
**Pull request by jameerpathan111**
This PR Adds Http proxy test scenario related to auto-attaching subscriptions to the content host.

Note: Manifest import, RH repo enable, and sync are also part of the test itself and that's why they are not part of the test setup.

### CODE REVIEW

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

File: tests/foreman/api/test_http_proxy.py [hunk 1]
```diff
@@ -125,3 +129,80 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+    rhel8_contenthost.install_katello_ca(module_target_sat)
+    rhel8_contenthost.register_contenthost(
+        org=function_org.name, activation_key=activation_key.name
+    )
+    assert rhel8_contenthost.subscribed
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_register(
+        {
+            'organization-id': function_org.id,
+            'content-view-id': content_view.id,
```
Comment (peterdragun) [inline tests/foreman/api/test_http_proxy.py:194]: using `cv` and `content_view` seems a bit confusing to me, in the end, it is the same content view. Please pick one and use it in the whole test.

File: tests/foreman/api/test_http_proxy.py [hunk 2]
```diff
@@ -125,3 +129,80 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+    )
+    host = module_target_sat.api.Host().search(
+        query={'search': f'name={rhel8_contenthost.hostname}'}
+    )[0]
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+    rhel8_contenthost.enable_repo(constants.REPOS['rhst7']['id'])
+    try:
+        rhel8_contenthost.execute('yum install -y zsh')
+    except ContentHostError:
```
Comment (peterdragun) [inline tests/foreman/api/test_http_proxy.py:207]: You don't need to catch this exception and then raise a new one without context

**Review by peterdragun — APPROVED** [review]
Left some non-blocking suggestions.

File: tests/foreman/api/test_http_proxy.py [hunk 3]
```diff
@@ -125,3 +129,80 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+            'lifecycle-environment-id': lce.id,
+            'name': rhel8_contenthost.hostname,
+        }
+    )
+    host = module_target_sat.api.Host().search(
+        query={'search': f'name={rhel8_contenthost.hostname}'}
+    )[0]
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+    rhel8_contenthost.enable_repo(constants.REPOS['rhst7']['id'])
```
Comment (ogajduse) [inline tests/foreman/api/test_http_proxy.py:204]: Enabling Sat tools for RHEL 7 on RHEL 8 system? :thinking: :monocle_face:

Comment (ogajduse) [reply tests/foreman/api/test_http_proxy.py:204]: Hmm, why do you even enable the repo if you do not use it?

**Review by ogajduse — CHANGES_REQUESTED** [review]

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

File: tests/foreman/api/test_http_proxy.py (hunk 3 above)
Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py]: not needed, just forgot to clean old code. Updated the pr.

File: tests/foreman/api/test_http_proxy.py (hunk 2 above)
Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py]: Updated the pr.

File: tests/foreman/api/test_http_proxy.py [hunk 4]
```diff
@@ -125,3 +129,80 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+    :Assignee: jpathan
+
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    with manifests.clone() as manifest:
+        upload_manifest(function_org.id, manifest.content)
```
[the remark this answers is not in the document]
Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py]: using manifester, let's see if prt passes.

File: tests/foreman/api/test_http_proxy.py [hunk 5]
```diff
@@ -125,3 +128,69 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    org = function_entitlement_manifest_org
+    with manifests.clone() as manifest:
+        upload_manifest(org.id, manifest.content)
```
Comment (synkd) [inline tests/foreman/api/test_http_proxy.py:165]:
```suggestion
```
Uploading the manifest is performed by the `function_entitlement_manifest_org` fixture, so it's not necessary to do so in the test body.

**Review by synkd — CHANGES_REQUESTED** [review]

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_http_proxy.py -k test_positive_auto_attach

File: tests/foreman/api/test_http_proxy.py [hunk 6]
```diff
@@ -125,3 +126,75 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+@pytest.mark.rhel_ver_match('8')
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize(
+    'setup_http_proxy',
+    [None, True, False],
+    indirect=True,
+    ids=['no_http_proxy', 'auth_http_proxy', 'unauth_http_proxy'],
+)
+@pytest.mark.tier3
+def test_positive_auto_attach(
```
Comment (adarshdubey-star) [inline tests/foreman/api/test_http_proxy.py:141]: Non blocking - test name can be more descriptive `test_positive_auto_attach_subscription(s)`

File: tests/foreman/api/test_http_proxy.py [hunk 7]
```diff
@@ -125,3 +126,75 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+            'lifecycle-environment-id': lce.id,
+            'name': rhel_contenthost.hostname,
+        }
+    )
+    host = module_target_sat.api.Host().search(
+        query={'search': f'name={rhel_contenthost.hostname}'}
+    )[0]
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+    rhel_contenthost.execute('yum install -y zsh')
```
Comment (adarshdubey-star) [inline tests/foreman/api/test_http_proxy.py:200]: What are we trying to achieve with this line of code? `rhel_contenthost.execute('yum install -y zsh')`

File: tests/foreman/api/test_http_proxy.py [hunk 8]
```diff
@@ -125,3 +126,75 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    org = function_entitlement_manifest_org
+    lce = module_target_sat.api.LifecycleEnvironment(organization=org).create()
+    content_view = module_target_sat.api.ContentView(organization=org).create()
+    rh_repo_id = enable_sync_redhat_repo(constants.REPOS['rhel8_bos'], org.id)
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_http_proxy.py:164]: why enable large rhel8_bos repo? As an alternative, would recommend enabling an infra_podman custom repo, or if a CDN repo is needed, try the ansible2.9 repo for auto-attach and install any package from that repo?

File: tests/foreman/api/test_http_proxy.py (hunk 6 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_http_proxy.py:141]: OR `test_positive_auto_attach_with_proxy()`

**Review by pondrejk — APPROVED** [review]
ack pending existing comments

File: tests/foreman/api/test_http_proxy.py (hunk 8 above)
Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py:164]: Historically there have been issues in syncing large repos via HTTP proxy, that's why I am testing rhel8_bos.

File: tests/foreman/api/test_http_proxy.py (hunk 7 above)
Comment (ogajduse) [reply tests/foreman/api/test_http_proxy.py:200]: @jameerpathan111 What I would like to see is assert that the package was installed. Either by checking `rc` of `execute` or running `rpm -q`.

Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py]: @adarshdubey-star yes, the package will be available from the subscription and we are checking if we can install it when HTTP proxy is configured on satellite.

Comment (jameerpathan111) [reply tests/foreman/api/test_http_proxy.py]: @ogajduse added assertion.

**Review by adarshdubey-star — APPROVED** [review]

File: tests/foreman/api/test_http_proxy.py [hunk 9]
```diff
@@ -125,3 +126,76 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
...
+            'name': rhel_contenthost.hostname,
+        }
+    )
+    host = module_target_sat.api.Host().search(
+        query={'search': f'name={rhel_contenthost.hostname}'}
+    )[0]
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+    result = rhel_contenthost.execute('yum install -y zsh')
+    assert result.status == 0
```
Comment (ogajduse) [inline tests/foreman/api/test_http_proxy.py:201]:
```suggestion
    assert result.status == 0, 'package was not installed'
```

**Review by ogajduse — APPROVED** [review]
One last, non-blocking comment.

**Pull request merged**